apioak/plugin/key-auth.lua

```diff
@@ -34,26 +37,25 @@ function _M.http_access(oak_ctx)
     if not oak_ctx.plugins[_M.key] then
         return false, nil
     end
-    --get key
-    local key = pdk.request.header('apikey')
-    if not key then
-        pdk.response.exit(401, { err_message = "Missing API key found in request" })
+
+    local plugin_conf = oak_ctx.plugins[_M.key]
+
+    local _, err = pdk.schema.check(schema, plugin_conf)
+    if err then
+        return false, nil
     end
```

Comment:
Data validation needs to be used when adding and does not need to be judged when requesting.